Sources/core/objects/Projectiles.cpp

```diff
@@ -0,0 +1,254 @@
+        }
+	else if(x<0 && y>0)
+        {
+	    //west
+            for(int e;x_moves>e;e++)
+            {
+            movement_.push_back(VD_LEFT);
+            }
+            //northwest
+            for(int i;diagonal_moves>i;i++)
+            {
+            movement_.push_back(VD_UPLeft);
+            }
+            return;	
+        }
+	else if(x<=0 && y<=0)
+        {
+	    //west
+            for(int e;x_moves>e;e++)
+            {
+            movement_.push_back(VD_LEFT);
+            }
+            //southwest
+            for(int i;diagonal_moves>i;i++)
+            {
+            movement_.push_back(VD_DOWNLeft);
+            }	
+            return;
+        }
+	
+	
+    }
+    else if(std::abs(x)<std::abs(y))
+    {
+        int y_moves = std::abs(y)-std::abs(x);
+	int diagonal_moves = std::abs(x);
+        if(x>=0 && y>=0)
+        {
+	    //north
+            for(int e;y_moves>e;e++)
+            {
+            movement_.push_back(VD_UP);
+            }
+            //north-east
+            for(int i;diagonal_moves>i;i++)
+            {
+            movement_.push_back(VD_UPRight);
+            }
+            return;	
+        }
+	else if(x>0 && y<0)
+        {
+	    //south
+            for(int e;y_moves>e;e++)
+            {
+            movement_.push_back(VD_DOWN);
+            }
+            //southeast
+            for(int i;diagonal_moves>i;i++)
+            {
+            movement_.push_back(VD_DOWNRight);
+            }
+            return;	
+        }
+	else if(x<0 && y>0)
+        {
+	    //north
+            for(int e;y_moves>e;e++)
+            {
+            movement_.push_back(VD_UP);
+            }
+            //northwest
+            for(int i;diagonal_moves>i;i++)
+            {
+            movement_.push_back(VD_UPLeft);
+            }
+            return;	
+        }
+	else if(x<=0 && y<=0)
+        {
+	    //south
+            for(int e;y_moves>e;e++)
+            {
+            movement_.push_back(VD_DOWN);
+            }
+            //southwest
+            for(int i;diagonal_moves>i;i++)
+            {
+            movement_.push_back(VD_DOWNLeft);
+            }
+            return;	
+        }
+    }
+    else if(std::abs(x)==std::abs(y))
+    {
+	if(x>=0 && y>=0)
+        {
+            //northeast
+            movement_.push_back(VD_UPRight);	
+            return;
+        }
+	else if(x>0 && y<0)
+        {
+            //southeast
+            movement_.push_back(VD_DOWNRight);
+            return;
+        }
+	else if(x<0 && y>0)
+        {
+            //northwest
+            movement_.push_back(VD_UPLeft);
+            return;
+        }
+	else if(x<=0 && y<=0)
+        {
+            //southwest
+            movement_.push_back(VD_DOWNLeft);
+            return;
+        }
+	
+    }
+    else
+    {
+    this->Delete();
+//never should go here but if it does we should delete that projectile
+    }
+}
+void Projectile::AfterWorldCreation()
+{
+    IMainObject::AfterWorldCreation();
```

Comment:
It is needed to call object direct base class (it should be `IMovable::AfterWorldCreation()`)